### PR TITLE
Jetpack plugin: Integrate Stats package

### PR DESCRIPTION
**Pull request by fgiannar**
This PR integrates the Stats package into the Jetpack plugin. The stats package (docs TBA in a follow up PR) is now the source of truth in terms of:
- Managing stats related options (see `Automattic\Jetpack\Stats\Options`)
- Building the tracking pixel data and adding it to the footer ( see `Automattic\Jetpack\Stats\Tracking_Pixel`)
- Registering the stats related XMLRPC endpoints (see `Automattic\Jetpack\Stats\XMLRPC_Provider`)
- Communicating with the WPCOM REST API Stats endpoints (see `Automattic\Jetpack\Stats\WPCOM_Stats`)

#### Changes proposed in this Pull Request:

**Adding the Stats package dependency**
* Updated the Jetpack plugin's `composer.json` with the Stats package dependency
* Updated the config package accordingly

* Managing stats options is now handled by `Automattic\Jetpack\Stats\Options`
**Stats options**
* deprecated `stats_get_options`
* deprecated `stats_get_option`
* deprecated `stats_set_options`
* deprecated `stats_set_option`
* deprecated `stats_upgrade_options`

**Tracking pixel**
* Building the tracking pixel data and adding it to the footer is now handled by `Automattic\Jetpack\Stats\Tracking_Pixel`
* removed `stats_array`
* deprecated `stats_add_to_footer`
* deprecated `stats_render_footer`
* deprecated `stats_render_amp_footer`
* deprecated `stats_build_view_data`

**XMLRPC endpoints**
* Stats related XMLRPC endpoints are now handled by `Automattic\Jetpack\Stats\XMLRPC`
* removed `stats_xmlrpc_methods`
* deprecated `stats_get_blog`
* removed `stats_esc_html_deep`

**Note:** Replacing the `stats_get_from_restapi` function in favor of `WPCOM_Stats` will be done in a follow-up PR as we are not deprecating that function yet and prefer not to overburden this one.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p9dueE-5Rd-p2

#### Testing instructions:

- Check the suggested test cases by the test-case-reminder bot
- Play around with the Jetpack plugin's stat settings and make sure they are respected
Screenshot 2022-10-06 at 09 56 14

- Make sure we track pageviews with [Infinite Scroll enabled](https://jetpack.com/support/infinite-scroll/)
- Make sure we track [Carousel](https://jetpack.com/support/carousel/) views 
- Pay extra attention to the tracking data we send and make sure the jp version is present as before (`j` param). As you might note the stats package is not longer adding this to the default data we are tracking, as its plugin agnostic. The Jetpack plugin is adding this parameter now via the `stats_array` filter
- Make sure to compare all the different places we show stats related data with `trunk` with the Stats module enabled AND disabled (you can disable the module by visiting `wp-admin/admin.php?page=jetpack_modules`):
- Site Stats in Jetpack dashboard
 Screenshot 2022-10-06 at 09 43 30   

- Site Stats page in Jetpack plugin
Screenshot 2022-10-06 at 09 46 21

- Stats widget in wp-admin home
Screenshot 2022-10-06 at 09 47 25

### CODE REVIEW

**Review by darssen — COMMENTED** [review]
I reviewed the code and everything looks good.
Prior to getting some more detailed test instructions, I tested the test-case-reminder steps and everything is working fine. Scripts are showing for amp and non-amp views. Also, Jetpack plugin Dashboard as well as wordpress.com/stats are displaying the views.

**Review by darssen — CHANGES_REQUESTED** [review]
I have been testing with the updated instructions and have the following comments

- Check the suggested test cases by the test-case-reminder bot  :heavy_check_mark: 

- Play around with the Jetpack plugin's stat settings and make sure they are respected  :heavy_check_mark:  (Edited)
    - <s>Pages viewed by an editor with the editor toggle enabled in the **Count logged in pages viewed from** section, are not being tracked. Info seems to be there in HTML but Stats are not being updated. It works fine in `trunk`</s> I tested again and cannot reproduce. Stats are being updated now. Might have been some kind of caching.
- Make sure we track pageviews with [Infinite Scroll enabled](https://jetpack.com/support/infinite-scroll/) :warning: When Infinite Scroll is enabled, stats are tracked. Not sure what is expected or if there is any specific stats to be checked here.
- Make sure we track [Carousel](https://jetpack.com/support/carousel/) views :warning:  When Carousel Views are enabled, stats are tracked. Not sure what is expected or if there is any specific stats to be checked here.
- Pay extra attention to the tracking data we send and make sure the jp version is present as before (j param). As you might note the stats package is not longer adding this to the default data we are tracking, as its plugin agnostic. The Jetpack plugin is adding this parameter now via the stats_array filter :heavy_check_mark: 
- Make sure to compare all the different places we show stats related data with trunk with the Stats module enabled AND disabled (you can disable the module by visiting wp-admin/admin.php?page=jetpack_modules): :heavy_check_mark:

File: projects/plugins/jetpack/3rd-party/class.jetpack-amp-support.php [hunk 1]
```diff
@@ -177,10 +178,12 @@ public static function disable_likes_admin_bar( $is_admin_bar_button_visible ) {
 	 * @since 6.2.1
 	 */
 	public static function add_stats_pixel() {
-		if ( ! has_action( 'wp_footer', 'stats_footer' ) ) {
+		if ( ! has_action( 'wp_footer', array( 'Automattic\Jetpack\Stats\Tracking_Pixel', 'add_to_footer' ) ) ) {
```
Comment (anomiex) [inline projects/plugins/jetpack/3rd-party/class.jetpack-amp-support.php]: You could make this a little clearer if you want by using `::class`.
```suggestion
		if ( ! has_action( 'wp_footer', array( Stats_Tracking_Pixel::class, 'add_to_footer' ) ) ) {
```

**Review by anomiex — COMMENTED** [review]
Looks ok from a code perspective. One inline comment, which may be ignored if you don't like it.

Haven't tested it myself, looks like someone else is already on that. 🙂

**Review by darssen — APPROVED** [review]
I was able to properly test the Carousel and I can see stats for the different images.
Infinite scroll testing is something that is evading me, but changes here should not impact it.

Amazing job!!

**Pull request merged**